src/tests/Microsoft.PowerFx.Core.Tests/IntellisenseTests/SuggestTest.cs

```diff
@@ -235,6 +237,26 @@ public void TestSuggestEmptyAll(string expression, params string[] expectedSugge
             Assert.Equal(expectedSuggestions, actualSuggestions);
         }
 
+        [Fact]
+        public void TestSuggestEscapedEnumName()
+        {
+            var enumStoreBuilder = new EnumStoreBuilder();
+            enumStoreBuilder.TestOnly_WithCustomEnum(new EnumSymbol(
+                new DName("Name That.Requires!escaping"),
```

Comment:
>! [](http://example.com/codeflow?start=45&length=1)

'.' please

"Name That.Requires.escaping"

Reply:
No, the ! is also a character that requires escaping. In this case it's not used as the ! operator, just as a character in an identifier.

Reply:
Oh sorry my bad. I didn't see that.